src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -252,17 +252,20 @@ void UDPEndPointImplLwIP::CloseImpl()
         mUDP              = nullptr;
         mLwIPEndPointType = LwIPEndPointType::Unknown;
 
-        // In case that there is a UDPEndPointImplLwIP::LwIPReceiveUDPMessage
+        // If there is a UDPEndPointImplLwIP::LwIPReceiveUDPMessage
         // event pending in the event queue (SystemLayer::ScheduleLambda), we
         // schedule a release call to the end of the queue, to ensure that the
         // queued pointer to UDPEndPointImplLwIP is not dangling.
-        Retain();
-        CHIP_ERROR err = GetSystemLayer().ScheduleLambda([this] { Release(); });
-        if (err != CHIP_NO_ERROR)
+        if (mDelayReleaseCount == 0)
```

Comment:
So this will do the delayed release only when there is no LwIPReceiveUDPMessage event pending.  And if the event is pending, this will just fail to release altogether.

That does not look right to me.